EDU-15536: Add info about single domain unification (#2034)
vtexdocs/dev-portal-content: docs/faststore/docs/go-live/1-configuring-external-dns.mdx

---

> title: '1. Configuring external DNS'

Reviewer: ```suggestion
title: '1. Configuring external DNS'
excerpt: 'Learn how to point your custom domain to FastStore and get your store ready for production.'
createdAt: '2024-01-01T00:00:00Z'
updatedAt: '2025-09-23T00:00:00Z'
```

---

> In this guide, you will learn how to configure external domains for your **FastStore** and **VTEX** website. The process involves the following steps:

Reviewer: ```suggestion
In this guide, you will configure your custom domain so it correctly points to your FastStore website hosted on [FastStore WebOps](https://developers.vtex.com/docs/guides/faststore/webops-dashboard) (`https://{accountName}.vtex.app/`).
Once the configuration is complete, customers will access your store using your final domain (e.g., `www.mystore.com`) instead of the default WebOps URL.
```

   - Configuring your custom domains in **VTEX Account Settings**.
   - Associating your custom domains with your **FastStore** project.

As soon as you complete this guide, your FastStore website will be accessible via a custom domain, and **VTEX Checkout** will also be operational. You will then be ready to continue configuring other VTEX solutions, such as **Order Placed**, **Login**, and **My Account**.

## Before you begin

   - Develop your FastStore project by following the [Getting Started track](https://developers.vtex.com/docs/guides/faststore/getting-started-overview).
   - Deploy your FastStore project.
   - Buy the desired domain.
   - [Redirect the **root domain** to the `www` subdomain.](https://help.vtex.com/en/tutorial/configuring-access-without-www--tutorials_4278#root-domain-redirect-without-www)

## Instructions

> ⚠️ If you still need to configure other VTEX solutions, **use a fictitious domain name** while following the steps. Once you have set up all the desired integrations, repeat the instructions using your final domain to make your website publicly available to end users.

> ### Step 1 - Creating DNS records

Reviewer: ```suggestion
### Step 1 - Creating DNS records

VTEX uses a CDN to serve FastStore websites. To make your domain reachable, you must create DNS CNAME records pointing your domain to VTEX’s CDN endpoints. Without this step, VTEX cannot validate or route traffic for your domain.

```

1. Access your domain provider's website.
2. Create a **CNAME** DNS record for your `www` domain (e.g., `www.mystore.com`) pointing to `www.{rootDomain}.cdn.vtex.com` (example: `www.mystore.com.cdn.vtex.com`). Check your domain provider's documentation for more information. The configured addresses may not be immediately available to everyone due to DNS propagation, which can take 24-48 hours to complete.

> ⚠️ If your strategy requires additional subdomains (e.g., `b2b.mystore.com`), add them as separate CNAMEs and ensure they are [listed under Hosts](#step3--configuring-domains-in-vtex-account-settings).  

### Step 2 - Putting your account into production

> To proceed, make sure your store is active and in production by following the steps in the guide [Putting your account into production](https://help.vtex.com/en/tracks/go-live-your-store--4Ns5FxIiksmjsdX2yOTduM/7wFsbWgN4rnZsbjhv8IItX#putting-your-account-into-production).

Reviewer: ```suggestion
Before VTEX can serve your website on a custom domain, your account must be configured as a production-ready environment. See [Putting your account into production](https://help.vtex.com/en/tracks/go-live-your-store--4Ns5FxIiksmjsdX2yOTduM/7wFsbWgN4rnZsbjhv8IItX#putting-your-account-into-production) for more information.
```

### Step 6 - Adding your store domain to FastStore WebOps

To finish the domain configuration, add your store domain to [FastStore WebOps](https://developers.vtex.com/docs/guides/faststore/webops-dashboard).

> ⚠️ FastStore WebOps domain configuration always takes priority over any settings in `discovery.config.js`. We recommend managing all domain configurations through FastStore WebOps to centralize all domain settings in a single source.

1. Access your [FastStore WebOps dashboard](https://developers.vtex.com/docs/guides/faststore/webops-dashboard).
2. Navigate to the **Settings** tab.
3. In the **Host** field, choose your store's domain (example: `www.mystore.com`) from the dropdown list.
4. Click `Add`.

   ![adding-host-webops](https://cdn.jsdelivr.net/gh/vtexdocs/dev-portal-content@main/docs/faststore/docs/go-live/adding-host-webops.png)

### Step 7 - Monitoring the new deployment

After adding your domain to FastStore WebOps, a new deployment will be triggered automatically. Before proceeding to the next step, monitor the deployment progress in the FastStore WebOps dashboard until it is completed.

> Then, check if your domain is configured correctly in the FastStore WebOps environment:

Reviewer: ```suggestion
```

6. Add the `secure` subdomain (example: `secure.mystore.com`) to the list.
   
   ![host-configuration](https://vtexhelp.vtexassets.com/assets/docs/src/account-settings-new___2e13bee3041e8505922a737b94652ca6.png)

7. Add the root domain as well; for example, if your domain is `www.mystore.com`, add `mystore.com`. For a complete setup, you must add three hosts for each domain:
   - `www.mystore.com` (store domain)
   - `mystore.com` (root domain)
   - `secure.mystore.com` (checkout/myaccount subdomain)

> ⚠️ To ensure a seamless checkout experience, all subdomains used by your store must be added to the host list. For more information, see the [Adding subdomains to the host list](#adding-subdomains-to-the-host-list) section.

#### Stores hosted on Vercel

Stores hosted on [Vercel](https://vercel.com/) can benefit from a single domain. This feature simplifies domain management by routing Checkout and My Account pages through the store’s primary domain (e.g., `www.domain.com`), removing the need for a separate `secure.domain.com` subdomain.

Add the hosts for the domain as shown in the example below:

   - `www.mystore.com` (store domain)
   - `mystore.com` (root domain)   

> </details>

Author: ```suggestion
```

4. In the **Hosts** field, enter your main domain (example: `www.mystore.com`) into the list.
5. Click `Add`.
6. Add the `secure` subdomain (example: `secure.mystore.com`) to the list.
   
   ![host-configuration](https://vtexhelp.vtexassets.com/assets/docs/src/account-settings-new___2e13bee3041e8505922a737b94652ca6.png)

7. Add the root domain as well; for example, if your domain is `www.mystore.com`, add `mystore.com`. For a complete setup, you must add three hosts for each domain:
   - `www.mystore.com` (store domain)
   - `mystore.com` (root domain)
   - `secure.mystore.com` (checkout/myaccount subdomain)

<details>
<summary><b>Stores hosted on Vercel</b></summary>

> Stores hosted on [Vercel](https://vercel.com/) can benefit from a single domain. This feature simplifies domain management by routing Checkout and My Account pages through the store’s primary domain (e.g., `www.domain.com`), removing the need for a separate `secure.domain.com` subdomain.

Author: ```suggestion
Stores hosted on [Vercel](https://vercel.com/) can use a single domain for all pages, including Checkout and My Account. This eliminates the need for a separate `secure.domain.com` subdomain (e.g., everything routes through `www.domain.com`).
```

> Add the hosts for the domain as shown in the example below:

Author: Since we are using this approach of a single domain for stores hosted on Vercel, are these instructions correct when setting up the hosts on VTEX Admin?

![single-domain-step3](https://github.com/user-attachments/assets/1d27b33a-a4f1-42b4-967f-60522f9e9479)

### Step 8 - Migrating the domain

1. Open a ticket with [VTEX Support](https://help.vtex.com/en/support) requesting a **domain migration** for your FastStore WebOps implementation. Include the following information:
   - VTEX account name
   - Store's domain (example: `www.mystore.com`)
   - Secure domain (example: `secure.mystore.com`)
   - Confirmation that you’ve completed all previous steps in this guide.
2. Wait for the VTEX team to process your request and complete the domain migration.
3. Once the migration is complete, test your store to ensure everything is working properly.

> ## Configuring custom domain prefixes

Reviewer: Mover essa seção para dentro do step 3 ou criar uma doc separada que possa ser usada como referência
